### PR TITLE
Add 256bit and 512bit vectorized hex decode

### PR DESCRIPTION
**Pull request by benaadams**
## Changes

- We already have a 128bit simd path; add 256bit and 512bit simd paths

#### What types of changes does your code introduce?

- [x] Optimization

#### Requires testing

- [x] Yes

#### If yes, did you write tests?

- [x] Yes

### CODE REVIEW

File: src/Nethermind/Nethermind.Core/Extensions/HexConverter.cs [hunk 1]
```diff
@@ -337,6 +337,134 @@ public static bool TryDecodeFromUtf8_Vector128(ReadOnlySpan<byte> hex, Span<byte
             return TryDecodeFromUtf8(hex[(int)offset..], bytes[(int)offset..], isOdd: false);
         }
 
+        public static bool TryDecodeFromUtf8_Vector256(ReadOnlySpan<byte> hex, Span<byte> bytes)
```
Comment (LukaszRozmej) [inline src/Nethermind/Nethermind.Core/Extensions/HexConverter.cs:340]: should this be internal?

**Review by LukaszRozmej — COMMENTED** [review]
Tests & Benchmarks?

File: src/Nethermind/Nethermind.Core/Extensions/HexConverter.cs (hunk 1 above)
Comment (benaadams) [reply src/Nethermind/Nethermind.Core/Extensions/HexConverter.cs]: Should be private; refactored

**Comment by benaadams** [comment]
| Method    | ByteLength | Mean       | Error     | StdDev    | Median     | Ratio | Alloc Ratio |
|---------- |----------- |-----------:|----------:|----------:|-----------:|------:|------------:|
| Scalar    | 32         |  32.720 ns | 0.8425 ns | 1.2349 ns |  32.698 ns |  1.00 |          NA |
| Vector128 | 32         |   5.188 ns | 0.2228 ns | 0.3334 ns |   5.134 ns |  0.16 |          NA |
| Vector256 | 32         |   3.229 ns | 0.0289 ns | 0.0423 ns |   3.234 ns |  0.10 |          NA |
| Vector512 | 32         |   3.009 ns | 0.1658 ns | 0.2431 ns |   3.133 ns |  0.09 |          NA |
|           |            |            |           |           |            |       |             |
| Scalar    | 64         |  56.427 ns | 0.3364 ns | 0.5035 ns |  56.424 ns |  1.00 |          NA |
| Vector128 | 64         |   7.598 ns | 0.0354 ns | 0.0530 ns |   7.599 ns |  0.13 |          NA |
| Vector256 | 64         |   4.812 ns | 0.0415 ns | 0.0608 ns |   4.807 ns |  0.09 |          NA |
| Vector512 | 64         |   4.123 ns | 0.1851 ns | 0.2655 ns |   4.134 ns |  0.07 |          NA |
|           |            |            |           |           |            |       |             |
| Scalar    | 128        | 107.334 ns | 0.4501 ns | 0.6310 ns | 107.531 ns |  1.00 |          NA |
| Vector128 | 128        |  13.617 ns | 0.0754 ns | 0.1105 ns |  13.625 ns |  0.13 |          NA |
| Vector256 | 128        |   7.892 ns | 0.1306 ns | 0.1787 ns |   7.840 ns |  0.07 |          NA |
| Vector512 | 128        |   6.161 ns | 0.1522 ns | 0.2182 ns |   6.157 ns |  0.06 |          NA |
|           |            |            |           |           |            |       |             |
| Scalar    | 256        | 211.792 ns | 0.6895 ns | 1.0320 ns | 211.823 ns |  1.00 |          NA |
| Vector128 | 256        |  25.945 ns | 0.0814 ns | 0.1193 ns |  25.958 ns |  0.12 |          NA |
| Vector256 | 256        |  14.105 ns | 0.0597 ns | 0.0893 ns |  14.090 ns |  0.07 |          NA |
| Vector512 | 256        |  11.926 ns | 0.0554 ns | 0.0830 ns |  11.913 ns |  0.06 |          NA |
|           |            |            |           |           |            |       |             |
| Scalar    | 512        | 423.199 ns | 2.1657 ns | 3.2415 ns | 422.230 ns |  1.00 |          NA |
| Vector128 | 512        |  50.393 ns | 0.4069 ns | 0.5964 ns |  50.362 ns |  0.12 |          NA |
| Vector256 | 512        |  27.170 ns | 0.1839 ns | 0.2753 ns |  27.170 ns |  0.06 |          NA |
| Vector512 | 512        |  23.254 ns | 0.3293 ns | 0.4827 ns |  22.974 ns |  0.05 |          NA |
|           |            |            |           |           |            |       |             |
| Scalar    | 1024       | 837.884 ns | 4.0460 ns | 5.9305 ns | 836.344 ns |  1.00 |          NA |
| Vector128 | 1024       | 102.157 ns | 0.3440 ns | 0.5149 ns | 102.196 ns |  0.12 |          NA |
| Vector256 | 1024       |  50.981 ns | 0.2298 ns | 0.3368 ns |  50.946 ns |  0.06 |          NA |
| Vector512 | 1024       |  46.550 ns | 0.0828 ns | 0.1240 ns |  46.521 ns |  0.06 |          NA |

**Comment by benaadams** [comment]
512 likely stronger on next version of Zen

**Review by LukaszRozmej — APPROVED** [review]

**Pull request merged**